src/Inkluzitron/Modules/Badges/BadgesModule.cs

```diff
@@ -0,0 +1,172 @@
+    {
+        private ReactionSettings ReactionSettings { get; }
+        private IHttpClientFactory HttpClientFactory { get; }
+        private DatabaseFactory DatabaseFactory { get; }
+        private BotDatabaseContext DbContext { get; set; }
+
+        public BadgesModule(DatabaseFactory databaseFactory, IHttpClientFactory httpClientFactory,
+            ReactionSettings reactionSettings)
+        {
+            DatabaseFactory = databaseFactory;
+            HttpClientFactory = httpClientFactory;
+            ReactionSettings = reactionSettings;
+        }
+
+        protected override void BeforeExecute(CommandInfo command)
+        {
+            DbContext = DatabaseFactory.Create();
+            base.BeforeExecute(command);
+        }
+
+        protected override void AfterExecute(CommandInfo command)
+        {
+            DbContext?.Dispose();
+            base.AfterExecute(command);
+        }
+
+        [Command("badge list")]
+        [Alias("badges")]
+        [Summary("Vypíše všechny odznaky a uživatele, kteří odznak získali.")]
+        public async Task ListBadgesAsync()
+        {
+            var msg = new StringBuilder();
+
+            var badges = DbContext.Badges.AsQueryable().Include(b => b.Users);
+            foreach (var badge in badges)
+            {
+                msg.AppendLine($"\n**{badge.Name} ({badge.Id})**");
+
+                if (badge.Description != null)
+                {
+                    msg.AppendLine($"*{badge.Description}*");
+                }
+
+                if (badge.Users.Count > 0)
+                {
+                    msg.AppendLine(string.Join(", ", badge.Users.Select(u => u.Name)));
+                }
+                else
+                {
+                    msg.AppendLine("*Tento odznak nikdo nezískal*");
+                }
+            }
+
+            if (msg.Length == 0)
+            {
+                msg.Append("*Nejsou k dispozici žádné odznaky.*");
+            }
+
+            await ReplyAsync(msg.ToString());
+        }
+
+        [Command("badge new")]
+        [Alias("badge create")]
+        [Summary("Vytvoří nový odznak. Za příkaz je třeba napsat id (bez mezery) a název odznaku. Na další řádek pak volitelně popis. Také je třeba přiložit obrázek odznaku (čtvercový, ideální velikost 64x64px).")]
+        public async Task CreateBadgeAsync(string id, [Remainder][Name("název")] string args)
+        {
+            var splitArgs = args.Split('\n', 2, StringSplitOptions.TrimEntries);
+
+            if (await DbContext.Badges.AsQueryable().AnyAsync(b => b.Id == id))
+            {
+                await ReplyAsync("Odznak s tímto id již existuje. Pro výpis všech odznaků použij `$badge list`.");
+                return;
+            }
+
+            var imageAttachement = Context.Message.Attachments.FirstOrDefault();
+            if (imageAttachement == null || !imageAttachement.Width.HasValue || imageAttachement.Width.Value != imageAttachement.Height.Value)
```

Comment:
attašemént baguette

Reply:
Typo fixed (pokud to byla narážka na to)